tests/pipeline/test_pipeline.py

```diff
@@ -444,6 +444,43 @@ def test_model_serialization(self, snapshot: SnapshotTest):
         enriched_pipeline = yaml.safe_load(result.stdout)
         snapshot.assert_match(enriched_pipeline, "test-pipeline")
 
+    def test_short_topic_definition(self):
+        result = runner.invoke(
+            app,
+            [
+                "generate",
+                "--pipeline-base-dir",
+                PIPELINE_BASE_DIR,
+                str(RESOURCE_PATH / "pipeline-with-short-topics/pipeline.yaml"),
+                "--defaults",
+                str(RESOURCE_PATH / "pipeline-with-short-topics"),
+            ],
+            catch_exceptions=False,
+        )
+
+        assert result.exit_code == 0
+
+        enriched_pipeline = yaml.safe_load(result.stdout)
+
+        assert isinstance(enriched_pipeline, dict)
```

Comment:
we should change this everywhere to use proper type annotations instead of runtime checks/narrowing
```suggestion
        enriched_pipeline: dict = yaml.safe_load(result.stdout)
```
but should be a followup